File: backend/app/schemas/conversation.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class ConversationSummary(BaseModel):
    id: UUID
    project_id: UUID
    title: str
    created_at: datetime
    updated_at: datetime


class ConversationMessage(BaseModel):
    id: UUID
    conversation_id: UUID
    question: str
    sql: str
    answer: str
    analysis_id: UUID | None = None
    queries_used: list[dict[str, Any]] | None = None
    created_at: datetime


class ConversationDetail(ConversationSummary):
    messages: list[ConversationMessage]

File: frontend/src/types/conversation.ts
import type { AnalysisRunResponse } from '@/types/analysis'

export type PlaygroundConversation = {
  id: string
  project_id: string
  title: string
  created_at: string
  updated_at: string
}

export type PlaygroundMessage = {
  id: string
  conversation_id: string
  question: string
  sql: string
  answer: string
  analysis_id: string | null
  queries_used: AnalysisRunResponse['queries_used'] | null
  created_at: string
}

export type PlaygroundConversationDetail = PlaygroundConversation & {
  messages: PlaygroundMessage[]
}
